src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2352,51 +2434,105 @@ private void ReportNullabilityMismatchWithTargetDelegate(SyntaxNode syntax, Name
             }
         }
 
-        private static TypeSymbolWithAnnotations InferResultNullability(BoundExpression operand, Conversion conversion, TypeSymbol targetType, TypeSymbolWithAnnotations operandType, bool fromConversionNode = false)
+        /// <summary>
+        /// Re-calculate and apply the conversion to the type of the operand and return the resulting type.
+        /// </summary>
+        private TypeSymbolWithAnnotations ApplyConversion(BoundExpression operand, Conversion conversion, TypeSymbol targetType, TypeSymbolWithAnnotations operandType)
         {
+            Debug.Assert(operand != null);
+            return ApplyConversion(operand, operand, conversion, targetType, operandType, checkConversion: true, fromExplicitCast: false, out _);
+        }
+
+        /// <summary>
+        /// Apply the conversion to the type of the operand and return the resulting type. (If the
+        /// operand does not have an explicit type, the operand expression is used for the type.)
+        /// If `checkConversion` is set, the incoming conversion is assumed to be from binding and
+        /// will be re-calculated, this time considering nullability. (Note that the conversion calculation
+        /// considers nested nullability only. The caller is responsible for checking the top-level nullability
+        /// of the type returned by this method.) `canConvert` is set if the conversion succeeded.
+        /// `node` is used only for the location of any diagnostics.
+        /// </summary>
+        private TypeSymbolWithAnnotations ApplyConversion(
+            BoundNode node,
+            BoundExpression operandOpt,
+            Conversion conversion,
+            TypeSymbol targetType,
+            TypeSymbolWithAnnotations operandType,
+            bool checkConversion,
+            bool fromExplicitCast,
+            out bool canConvert)
```

Comment:
>canConvert [](start = 21, length = 10)

This is probably more properly called `nullabilityMismatch`, as we know the convertion is already OK from binding's point of view.

Reply:
Renamed to `canConvertNestedNullability`.

---
In reply to: [190760361](https://github.com/dotnet/roslyn/pull/26356#discussion_r190760361) [](ancestors = 190760361)